crates/katana/rpc/rpc-types/src/simulation.rs

```diff
@@ -0,0 +1,41 @@
+use anyhow::Result;
+use katana_primitives::block::BlockIdOrTag;
+use katana_primitives::chain::ChainId;
+use katana_primitives::transaction::{ExecutableTx, ExecutableTxWithHash};
+use serde::{Deserialize, Serialize};
+
+use crate::transaction::BroadcastedTx;
+use crate::SimulationFlag;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SimulateTransactionsRequest {
+    pub block_id: BlockIdOrTag,
+    pub transactions: Vec<BroadcastedTx>,
+    pub simulation_flags: Vec<SimulationFlag>,
+}
+
+impl SimulateTransactionsRequest {
+    /// Converts the request into a vector of executable transactions.
+    pub fn try_into_executables_with_hash(
```

Comment:
Dont need a specific method for this. lets just call the `.into_tx_with_chain_id` on the individual `Vec<Broadcasted>` directly like how we did for `estimate_fee`.